streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -538,13 +539,18 @@ public void flush() {
                 } catch (final RuntimeException exception) {
                     if (firstException == null) {
                         // do NOT wrap the error if it is actually caused by Streams itself
-                        if (exception instanceof StreamsException)
+                        // In case of FailedProcessingException Do not keep the failed processing exception in the stack trace
+                        if (exception instanceof FailedProcessingException)
+                            firstException = new ProcessorStateException(exception.getCause());
```

Comment:
IMO, we should keep it exactly the same as before the `FailedProcessingException` was introduced:
```suggestion
                            firstException = new ProcessorStateException(
                                format("%sFailed to flush state store %s", logPrefix, store.name()), exception.getCause());
```

Reply:
done